### ISSUE TITLE
[Bug] Cursor becomes misaligned when using custom font on DESKTOP

### ISSUE THREAD

**Issue by thejupitergroup** [microsoft/monaco-editor#4644]
### Reproducible in vscode.dev or in VS Code Desktop?

- [X] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [ ] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Monaco Editor Playground Link

[The Playground Link](https://microsoft.github.io/monaco-editor/playground.html?source=v0.50.0#XQAAAALVBgAAAAAAAABBqQkHQ5Njdzot23jh63LCu7UJx9l34ba83hIon8WtqyPMkFIdW4pKVhWCfVWkrH_ODZ176Xw0UzgWogOkJxMjnFpQPRXuSmRrULUxyUM_cMwnFAHgptj6uyMvz9cPLAxzLHUW53JP86kd6N77di27G0M0nm0kYGp2Xx-md3q-p-p8yOK8BrXOoPkmGmEq6zR531I49-4fnfaVip2hn7D5UGkt2maN_SCmh3RqdrYbpKVGHmtRRHFZ2J7aT0c3Kj9fUfZRWDg2jhFMl4C-j6rz5j-DsVlEW80mt8iOb1452Aq2sJ3DVLlcuBgrw4scL9ox2tWGCcxtnO6mnVcx-7o0MllNLUkRGJpO0KqAfvClE8oOaABAzirxfX9Zq2ecn1V2AUdVZfKQwObhn1WXxljsJs4jD-lMqnYP7vsQCMXXf5C7eiSdHFkbBkvcavrgkg6DvAtNggM9NUTEdx3P885U1KPg7rpNLEqoKfoweMZTyK8MOwSJ27YKDdJhCaU_3g0toliINwSVgdNR8E-iFbt2EldK7fFku9xa_Ok-KZCW76FItQTisrcI7FVJDJVTA3t17q7v0_gnLbuiKOY-lasjTIL5cFCBikDrFPExsfQ81Ta4OatVFwFcwWRhJ6ClijdcM_Kmr1dECa2oOYImlCFUcNYcoHpsmHKwYcXXW6nhFTGNUGxSE2qqiaOslra0F3t4AaWFvJ5vMLpXyzDvh-w1JPcPCtzC1lWNjIwwOqXUzNzC6mECixA7QogKBsyFABzfDKBB5O6HUKyKOrLtxZPYFP8MM4t6vr54iEXJl53wu9SKODd0aQdmTqdv6t-M6RnC_KU8wNlkN-Az87Itm3a57cMyxBvPJjfKGdbpzLBFiRjd3ZIaFcGb3g68-7wIk_nNvcaAkB5rXzAMHjqN-As-P58r6RE54qg4cBm7prhANqrsBaIN2DCXuCJ_iSHzonjVR8DN8YCfxvwtQovAVcgX1jU2n_u-dSNsDfTefx0835o9Krf0N4lmTMjDap4tvL-Nj26vsIQz1gzm-_6QCmE3v8MV00vQduPm4jSUF5CPahF0C9uEx8laRIiHkh-oLM4JDb4CLltyLf-KvjsA)

### Monaco Editor Playground Code

```typescript
JS: 
require.config({ paths: { 'vs': 'monaco-editor/min/vs' } });

require(['vs/editor/editor.main'], function () {
    const editor = monaco.editor.create(document.getElementById('editor'), {
        value: '<!DOCTYPE html>\n<html lang="en">\n<head>\n    <meta charset="UTF-8">\n    <meta name="viewport" content="width=device-width, initial-scale=1.0">\n    <title>Document</title>\n</head>\n<body>\n    <h1>Hello, Monaco Editor!</h1>\n</body>\n</html>',
        language: 'html',
        theme: 'vs-dark',
        automaticLayout: true,
        fontFamily: 'Laurentia Code',
    });

    const preview = document.getElementById('preview');

    editor.onDidChangeModelContent(() => {
        const html = editor.getValue();
        preview.srcdoc = html;
    });
});

HTML:
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>CodePen-like Editor</title>
    <link rel="stylesheet" href="style.css">
</head>
<body>
    <div id="container">
        <div id="editor"></div>
        <iframe id="preview" sandbox="allow-scripts"></iframe>
    </div>
    <script src="monaco-editor/min/vs/loader.js"></script>
    <script src="script.js"></script>
</body>
</html>

CSS:
@font-face {
    font-family: 'Laurentia Code';
    src: url(https://andorraeditor.pages.dev/DejaVuSansMono.ttf);
}

body, html {
    margin: 0;
    padding: 0;
    height: 100%;
    display: flex;
    flex-direction: column;
    font-family: Arial, sans-serif;
}

#container {
    display: flex;
    height: 100%;
    width: 100%;
}

#editor {
    width: 50%;
    height: 100%;
}

#preview {
    width: 50%;
    height: 100%;
    border: none;
}
```

### Reproduction Steps

We were loading a custom font a code editor we wanted to create. When we entered the URL and changed the font in the CSS file, the cursor would be gradually misplaced in the Monaco editor, both in the playground and on our website.

### Actual (Problematic) Behavior

When the custom font gets loaded, as the user moves right in a line (or farther away from the left size of the editor) their cursor would start to become misaligned. This can only be replicated on desktop (Win11 for us). We have tried a few other fonts and the 
text cursor still does the same thing.

### Expected Behavior

The cursor should stay aligned with the text, no matter what font.

**Comment by neehhaa06** [comment]
Hello @thejupitergroup :)

> Make sure the font you've created is a real monospace font. Next, in the Monaco Editor options, modify letterSpacing or turn off fontLigatures to correct the alignment of the cursor:         

> If the issue persists, switch to a reliable monospace font like "Consolas."

**Comment by Suyious** [comment]
Any updates on this bug?

**Comment by thejupitergroup** [comment]
Yeah, sorry. We forgot to close this. You can use this function in the code to make sure the font cache gets reset.

`function ensureFontsLoaded(callback) {
        document.fonts.ready.then(() => {
            monaco.editor.remeasureFonts();
            callback();
        });
    }

    ensureFontsLoaded(() => {
        console.log("Fonts loaded and remeasured");
    });`
